File: server/models/fabricators/Fabricator.py
from collections import deque
import datetime


class Fabricator:
    def __init__(
        self,
        id: int,
        hardware_id: str,
        custom_name: str,
        date_registered: datetime,
        model_name: str,
        device_file: str,
        job_queue: deque,
    ):
        self.id = id
        self.hardware_id = hardware_id
        self.custom_name = custom_name
        self.date_registered = date_registered
        self.model_name = model_name
        self.device_file = device_file
        self.job_queue = job_queue

    def __repr__(self):
        return (
            f"id: {self.id}, hardware_id: {self.hardware_id}, custom_name: {self.custom_name},"
            f"date_registered: {self.date_registered}, model_name: {self.model_name}, "
            f"device_file: {self.device_file}, job_queue: {self.job_queue}"
        )

File: client/src/models/Fabricator.ts
import { ref } from 'vue'

// Keeps track of all registered fabricators
export const fabrictorList = ref<Fabricator[]>([])

// Represents the Fabricator object
export interface Fabricator 

{
	id? : number
	hardwareId? : string
	customName : string
    description? : string
	dateRegistered: number
	modelName : string
	jobQueue : Job[]
	isQueueExpanded : boolean
	isWaitingForResponse : boolean
}

// Sets the fabricator information, adds a Fabricator to the fabricatorList, and returns a Fabricator
export function registerFabricator(userSubmission : {customName: string; description: string; modelName: string, date: number;}) : Fabricator
{
	const newFabricator : Fabricator = {
		customName: userSubmission.customName,
		description: userSubmission.description,
		dateRegistered: userSubmission.date,
		modelName: userSubmission.modelName,
		jobQueue: [],
		isQueueExpanded : false,
		isWaitingForResponse : false

	}
	fabrictorList.value.push(newFabricator)
	return newFabricator
}

/*
WIP
add_job(fabricator_id : number, job : Job)
remove_job(fabricator_id : number, job_id : number)
get_job_status(fabricator_id: number)
edit_job_status(fabricator_id: number, job_status : string)
move_job(fabricator_id: number, job_id : number, new_position : number)
edit_name(fabricator_id: number, custom_name : string)
rerun_job(fabricator_id: number, job_id: number)
start_job(fabricator_id: number)
get_registered()
get_all()
*/




